app/controllers/sensors_controller.rb

```diff
@@ -15,15 +15,9 @@ def show
   end
 
   def index
-    @sensors = Sensor.includes(:measurements).all
+    @sensors = Sensor.includes(:locations).all
 
-    render json: @sensors
-  end
-
-  def new
-    @sensor = Sensor.new
-
-    render partial: 'sensors/new'
+    render json: @sensors, include: :locations, except: [:token, :created_at, :updated_at]
```

Comment:
You can limit also location keys returned in JSON (see http://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html for examples).